src/routes/routes.json

```diff
@@ -5157,6 +5187,11 @@
           "type": "string"
         }
       },
+      "produces": [
+        "application/json",
+        "multipart/related",
+        "multipart/form-data"
```

Comment:
How about these endpoint with multiple `produces`?

```json
"produces": [
  "application/json",
  "multipart/related",
  "multipart/form-data"
]
```

Can these endpoints actually return different content-types? :thinking: Or are these just errors in Bitbucket's swagger specification? Can you please take a look at this?

Reply:
yeah, I'm pretty sure the swagger is messed up.  The /snippet endpoint has POST, PUT, and GET.

The POST doesn't have and produces or consumes.
The PUT and GET both have:

```
"produces": [
          "application/json",
          "multipart/related",
          "multipart/form-data"
        ],
```

which is wrong as the descript for the GET says it will only return json.
I'm guessing it's supposed to be **consumes** and is supposed to be on PUT and POST, not GET.

At any rate, we need to use the pipelines/log endpoint which correctly has:

```
"produces": [
          "application/octet-stream"
        ],
```

and we need that to be sent as the accept header.  This was the quickest and so far only way we could see how to do it.

Reply:
@rundquist @jdoklovic 

I can propose a cleaner solution:

- In `generate-routes.js` if `produces.length === 1` add the following `headers` object in `routes.json`:

`{ "accept": produces[0] }`

Resulting:
```json
...
"getPipelineStepLog": {
  ...
  "headers": {
    "accept": "application/octet-stream"
  },
  ...
}
...
```

That's it.

Reply:
@MunifTanjim can go ahead and do it?  also, can you give us the version in npm when it's published?

Reply:
@MunifTanjim in case you haven't seen, we're using it for this: https://marketplace.visualstudio.com/items?itemName=Atlassian.atlascode

Reply:
@MunifTanjim I just pushed an update. I'm filtering out everything that isn't `octet-stream` in `generate-routes` and then using the value if it exists in `request-options`. This way future changes to the swagger shouldn't cause issues.

Reply:
@rundquist Thanks! I've tweaked it a little bit to make it more general.

For reference, here's the current output:

```js
bitbucket.authenticate({
  type: 'apppassword',
  username: 'MunifTanjim',
  password: '12345'
})

bitbucket.repositories.getPipelineStepLog({
  username: 'MunifTanjim',
  repo_slug: 'test',
  pipeline_uuid: '12345',
  step_uuid: '12345'
})
```

```sh
$ DEBUG=bitbucket node bitbucket-test.js 
  bitbucket REQUEST: { timeout: 0,
  method: 'GET',
  url:
   'https://api.bitbucket.org/2.0/repositories/MunifTanjim/test/pipelines/12345/steps/12345/log',
  headers:
   { accept: 'application/octet-stream',
     'user-agent': 'NodeBitbucket/1.10.0',
     authorization: 'Basic TXVuaWZUYW5qaW06MTIzNDU=' },
  body: undefined }
```

The `headers.accept` is being set correctly.